### ISSUE TITLE
Low quality of pose labels on STD dataset

### ISSUE THREAD

**Issue by shanice-l** [PKU-EPIC/DREDS#5]
Hi, authors!

Thanks for your great work!

Because you didn't offer the pose label of the STD dataset, I tried to calculate the ground-truth pose by the Umeyama algorithm.
I've found that the calculated pose quality is low, just like the picture below shows.
![image](https://user-images.githubusercontent.com/52108026/188461835-1ea2b9a1-1da6-4822-aadf-f6d5845f5aac.png)

I'm wondering if you can release the ground-truth label of the STD dataset, or it would be hard to follow your job.

Looking forward to your reply!

**Comment by shanice-l** [comment]
BTW, the algorithm I used to calculate the pose was borrowed from your inference code. So I'm doubting the accuracy of pose metrics as well.

**Comment by Jiyao06** [comment]
Hi,
Thanks a lot for your interest in our work!

We provide the ground truth depth(xxx_gt_depth.exr) and  nocs map(xxx_coord.png) for pose fitting using Umeyama.  As shown in the following, the label of objects in STD is accuracy.

<img width="461" alt="image" src="https://user-images.githubusercontent.com/88179466/188557030-49477c27-1a0e-40cc-8fe7-30ae8e076833.png">

**Comment by shanice-l** [comment]
Thanks for your timely reply!

Maybe there are some mistakes in my code. Could you please offer me the code to generate pose labels as in the picture?

**Comment by Jiyao06** [comment]
Hi,
Thanks a lot for your interest in our work!

We provided the ground truth depth(xxx_gt_depth.exr) and nocs map(xxx_coord.png) for pose fitting using Umeyama. As shown in the following, the label of objects in STD is accurate.

<img width="466" alt="image" src="https://user-images.githubusercontent.com/88179466/188557949-01938ee1-d703-4b71-abe0-7d83d1efa7c6.png">

**Comment by shanice-l** [comment]
As a reference, my code for generating pose label is shown below

```python

def annotate_data(data_dir, splits):
    for split in splits:
        image_lists = open(os.path.join(data_dir, f"image_set/{split}_list.txt")).read().splitlines()
        scaled_intrinsics = intrinsics = np.array([[918.295227050781, 0, 639.5], [0, 917.5439453125, 359.5], [0, 0, 1]])
        # resize image
        scaled_intrinsics[:2] = scaled_intrinsics[:2] / 2

        for ind in image_lists:
            image_path = osp.join(data_dir, f"{ind}_color.png")
            nocs_path = osp.join(data_dir, f"{ind}_coord.png")
            gt_depth_path = osp.join(data_dir, f"{ind}_gt_depth.exr")
            sim_depth_path = osp.join(data_dir, f"{ind}_depth_415.exr")
            mask_path = osp.join(data_dir, f"{ind}_mask.png")
            meta_info_path = osp.join(data_dir, f"{ind}_meta.txt")
            meta_infos = load_meta(meta_info_path)

            image = read_image_mmcv(image_path, format="BGR")   # (720, 1280, 3)
            resize_image = cv2.resize(image, (640, 360), interpolation=cv2.INTER_NEAREST)
            nocs = mmcv.imread(nocs_path, "unchanged") / 255.0  # (360, 640, 3)
            _syn_depth = cv2.imread(gt_depth_path, cv2.IMREAD_ANYCOLOR | cv2.IMREAD_ANYDEPTH)
            # _sim_depth = cv2.imread(sim_depth_path, cv2.IMREAD_ANYCOLOR | cv2.IMREAD_ANYDEPTH)
            # _sim_depth = cv2.resize(_sim_depth, (640, 360), interpolation=cv2.INTER_NEAREST)
            if len(_syn_depth.shape) == 3:
                _syn_depth = _syn_depth[:, :, 0]  # (h, w)
    
            _mask = mmcv.imread(mask_path)[:, :, 0]  # (h, w)
            _mask_sem = np.full(_mask.shape, 8)
            _scale = np.ones((8, 3))  # (class_num+1,3)

            for meta in meta_infos:
                index = meta["index"]
                label = meta["label"]
                cat_name = meta["instance_folder"]
                inst_id = meta["name"]
                # scale = meta["scale"]

                _mask_sem[_mask == index] = label
                if cat_name != "":
                    bbox_path = osp.join(PROJ_DIR, "datasets/cad_model/real_cat_known/", cat_name, inst_id, "bbox.txt")
                    bbox = np.loadtxt(bbox_path) # (2, 3)
                    _scale[label] = bbox[0] - bbox[1]
                else:
                    _scale[label] = np.ones((3))
                _scale[label] /= np.linalg.norm(_scale[label])

            gt_class_ids, gt_scores, gt_masks, gt_coords, gt_boxes = prepare_data_posefitting(
                _mask_sem, nocs
            )
            gt_RTs, scales, error_message, _ = align(
                gt_class_ids, gt_masks, gt_coords, _syn_depth, scaled_intrinsics, synset_names,
            )
            bbox3d_list = [get_bbox_from_scale(scale) for scale in scales]
            bbox2d_list = [misc.project_pts(bbox3d, scaled_intrinsics, R, t) for (bbox3d, R, t) in zip(bbox3d_list, gt_RTs[:, :3, :3], gt_RTs[:, :3, 3])]
            for bbox2d, gt_class_id in zip(bbox2d_list, gt_class_ids):
                obj_name = synset_names[gt_class_id]
                resize_image = misc.draw_projected_box3d(resize_image, bbox2d)

                plt.imshow(resize_image[:, :, ::-1])
                plt.title(obj_name)
                plt.show()
```

**Comment by Jiyao06** [comment]
We provided a [script](https://github.com/PKU-EPIC/DREDS/blob/main/CatePoseEstimation/align.py) to generate the GT Pose that you can refer to. Thanks!
